flang/include/flang/Optimizer/Dialect/FIROps.td

```diff
@@ -74,6 +74,10 @@ def fir_BoxProcType : Type<CPred<"$_self.isa<fir::BoxProcType>()">,
 def AnyBoxLike : TypeConstraint<Or<[fir_BoxType.predicate,
     fir_BoxCharType.predicate, fir_BoxProcType.predicate]>, "any box">;
 
+// Reference and Box like types
+def AnyRefOrBoxLike : TypeConstraint<Or<[AnyReferenceLike.predicate, AnyBoxLike.predicate]>,
+    "any reference or box like">;
+
```

Comment:
Was this, or something like it, added in another PR recently?

Reply:
I recently added AnyShapeOrShift, but I guess that's not what you mean. I also recently added `fir::dyn_cast_ptrOrBoxEleTy` that can be used to unwrap things that would answer true to this new predicate.